core/src/main/java/com/linecorp/armeria/server/http/healthcheck/ManagedHttpHealthCheckService.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.healthcheck;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import com.google.common.base.Ascii;
+
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.http.AggregatedHttpMessage;
+import com.linecorp.armeria.common.http.HttpData;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponseWriter;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * A {@link HttpHealthCheckService} which allows overriding its status via a POST request.
+ * <p>
+ * <h2>Example</h2>
+ * <pre>{@code
+ * Server server = new ServerBuilder().serviceUnder("/health", new ManagedHttpHealthCheckService())
+ *                                    .build()
+ * }</pre>
+ * Default config uses content on or off
+ * You can also use your own path matching.
+ * </p>
+ * <pre>{@code
+ * ...
+ * Server server = new ServerBuilder().serviceUnder("health", new ManagedHttpHealthCheckService() {
+ *         {@literal @}Override
+ *         public CompletableFuture&lt;Optional&lt;Boolean&gt;&gt; mode(HttpRequest req) {
+ *             return CompletableFuture.completedFuture(Optional.empty());
+ *         }
+ *     }).build())
+ * ...
+ * }</pre>
```

Comment:
The `{@code ..}` is equivalent to `<code>{@literal}</code>`, so it negates HTML tags.
The `{@code ..}`, also does not work properly when `@` starts a line in the tag like this
    {@code ...
    ...
    @myTag
    }
The @ character should be ignored and printed literally because it is in `{@code ..}`, but it is interpreted as another javadoc tag.

So this is my suggestion. 
1. use `{@code ..} when there is no `@` character that starts the line.
2. use <code></code> with `&#64;` when there is `@` character that starts the line. Using {@literal @} makes confusing because there is a space that I can’t remove in front of `@`.
3. use {@literal ..} for other characters, in above situation.
So, here is the example for this javadoc.

    <pre><code>
     ...
    Server server = new ServerBuilder().serviceUnder("health", new ManagedHttpHealthCheckService() {
        &#64;Override
        public{@literal CompletableFuture<Optional<Boolean>>} mode(HttpRequest req) {
            return CompletableFuture.completedFuture(Optional.empty());
        }
    }).build())
    ...
    }</code></pre>

@trustin what do you think?